### PR TITLE
Feature/Web Engine support

### PR DESCRIPTION
**Pull request by dboltovskyi**
Implements [#182](https://github.com/smartdevicelink/sdl_atf/issues/182)

This PR is **ready** for review.

### Summary
Current PR contains implementation of WebSocket client for Mobile Connections

### Changelog
 - Implemented Mobile Adapter Controller
 - Implemented WebSocket mobile adapter
 - Implemented WSS support
 - Introduced mobile configuration with default `TCP` value

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by mrapitis** [comment]
@theresalech Ford has approved of this PR.

File: README.md [hunk 1]
```diff
@@ -131,6 +132,10 @@ config.hmiUrl = "ws://localhost"
 config.hmiPort = 8087
 config.mobileHost = "localhost"
 config.mobilePort = 12345
+config.wsMobileURL = "ws://localhost"
+config.wsMobilePort = 2020
+config.wssMobileURL = "wss://localhost"
```
Comment (JackLivio) [inline README.md]: Copy and paste error that this was added twice?

File: README.md [hunk 2]
```diff
@@ -31,30 +31,31 @@ $ git submodule update
...
 
-**2** Install Qt5.3+
-
-[Download Qt5.3+](https://download.qt.io/archive/qt/5.3/5.3.1/)
+**2** Install Qt5.9+
+- Run the following commands :
 ```
-sudo chmod +x some-qt-build.run
-./some-qt-build.run
+$ sudo add-apt-repository -y ppa:beineri/opt-qt591-xenial
```
Comment (JackLivio) [inline README.md]: These steps do not work for ubuntu 18 Bionic

Comment (dboltovskyi) [reply README.md]: @JackLivio Yes, you're correct. This update is applicable for `16.04`. For `18.04` the following should work:
```
qt5-default libqt5websockets5-dev
```

Comment (aderiabin) [reply README.md]: @JackLivio Instruction has been updated accordingly (see 17251cf)

**Comment by dboltovskyi** [comment]
@JackLivio Please notice:
1)  conflicts were resolved in 25e5c45 and 8080699
2) new `deviceId` handling logic for WebEngine device was implemented in b4c35f4

**Comment by AByzhynar** [comment]
@JackLivio Please re-approve this PR if you are ok with new proposed changes. Thank you.

**Comment by aderiabin** [comment]
@JackLivio Please notice:
 - Control of minimal QT version added into cmake file in a41fb2e

**Pull request merged**